src/PimApi/PimCatalogApiClient.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\PimApi;
+
+use App\PimApi\Model\Catalog;
+use App\Storage\AccessTokenStorageInterface;
+use App\Storage\PimURLStorageInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+class PimCatalogApiClient
+{
+    public function __construct(
+        private HttpClientInterface $client,
+        private readonly PimURLStorageInterface $pimURLStorage,
+        AccessTokenStorageInterface $accessTokenStorage,
+    ) {
+        $this->client = $this->client->withOptions([
+            'headers' => [
+                'Content-type' => 'application/json',
+                'Authorization' => 'Bearer '.$accessTokenStorage->getAccessToken(),
+            ],
+        ]);
+    }
+
+    public function getCatalog(string $catalogId): Catalog
+    {
+        $pimUrl = $this->pimURLStorage->getPimURL();
```

Comment:
```suggestion
        $pimUrl = $this->pimURLStorage->getPimURL();
```

this can return null.
You should always check that is not null before concatenating in the URL you are calling.
In similar places, we throw LogicException when that happens.
I wonder, why did phpstan/psalm was not complaining about it in your PR ?